### PR TITLE
Move minimum supported Python version to 3.7

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

### CODE REVIEW

File: CHANGE_LOG [hunk 1]
```diff
@@ -3,6 +3,10 @@ Version 0.54.0
 
 In development
 
+NOTES:
+
+* Minimum python version raised to 3.7.
```
Comment (sklam) [inline CHANGE_LOG:8]: We should do this more often. Leaving notes in the changelog for important changes so we don't have to recall 3mos worth of work at release time.

**Review by sklam — APPROVED** [review]
Thanks for the patch!

**Comment by sklam** [comment]
@esc, I think we'll need to sync buildfarm before merging this.

**Comment by esc** [comment]
BFID: `numba_smoketest_cpu_yaml` w/o Python 3.6 tasks.

**Comment by esc** [comment]
`numba_smoketest_cpu_yaml_17` was fine.

**Comment by esc** [comment]
Thank you for the patch.

**Pull request merged**